### PR TITLE
Add Text Alignment Support to the Pullquote Block

### PR DESCRIPTION
**Pull request by porada**
## Description

This PR adds support for specifying text alignment inside the Pullquote block using a block toolbar control.

Our users expect the center alignment option to be available (#22784, #31558, Automattic/wp-calypso#52519), because it used to also align the text inside the quote to center. That option was removed probably with the introduction of the Solid Color block style which aligned text to the side. This PR allows the user to overwrite the default alignment defined by any of the available block styles.

Behind the scenes, this PR also adds support for the `placeholder` attribute in the standard `AlignmentToolbar` control. This allows the control to reflect the currently used setting without explicitly setting the value. 

## How has this been tested?

* Pull the changes.
* `npm run test-unit`
* Open the editor.
* Insert an instance of the Pullquote block.
* Set the block style to Solid Color.
* Use the new toolbar control to overwrite the default text alignment.
* Make sure the custom text alignment option works intact with any other block alignment and block style option available—both in the editor and on the preview.

## Types of changes

New feature

## Checklist

- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows the accessibility standards.
- [x] I’ve tested my changes with keyboard and screen readers.
- [x] My code has proper inline documentation.
- [x] I’ve included developer documentation if appropriate.
- [ ] I’ve updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Comment by porada** [comment]
Just making a note that the failing check keeps encountering a timeout. The suite should run just fine locally.

Also, please let me know if I should turn 8225d471f841d1cd65a4541331301037fc13c5e1 into a separate PR.

File: packages/block-library/src/pullquote/style.scss [hunk 1]
```diff
@@ -30,15 +31,32 @@
...
+/*
+ * 1. Defining `text-align` on the parent would get overwritten by the theme’s
+*     defintion of the default style.
+ * 2. When no text alignment is set, the solid color style aligns text to the
+ *    side, not center.
+ * 3. Twenty Twenty-One has a hardcoded `text-align` on `blockquote::before`.
+ *    The only way to make the decorative quotation mark follow the same
+ *    alignment as its parent (both in the editor and on the front-end) is
+ *    by resetting the property with `!important`.
+ */
```
Comment (yansern) [inline packages/block-library/src/pullquote/style.scss:43]: Excellent CSS commenting style. 👍

**Review by yansern — APPROVED** [review]
I am testing this on the latest rebased 10.7.1 version.

Unit tests are passing. (There was one failing test unrelated to this PR.)
image

Text alignment on pullquote block works as expected.
image

IMO the codes are clean & excellent.

File: packages/block-library/src/pullquote/style.scss (hunk 1 above)
Comment (draganescu) [reply packages/block-library/src/pullquote/style.scss:43]: Plus one to that! 👏🏻

**Comment by porada** [comment]
I can the see the PR has diverged a bit from the trunk over time. Once there is a clear desire to merge this in, I’ll be happy to resolve all the conflicts. Just let me know.

**Comment by SiobhyB** [comment]
👋 @porada, I did some testing to double-check whether this change has any impact on the iOS and Android apps. I didn't run into issues, but there have been some updates to block settings in the apps in the last few months that aren't included in this PR (more specifically, the introduction of global styles). It'd be great to test again when the PR's updated from trunk, so that we can confirm that all the settings still work as expected on mobile. 

Would you be able to give me a ping whenever this PR's updated? I'll follow along with this issue for any updates, too. Thanks!

**Comment by Mamaduka** [comment]
Hi, @porada

Can you rebase this branch on top of the current trunk?

Thanks

**Comment by carolinan** [comment]
Text align was added to the block in https://github.com/WordPress/gutenberg/pull/30951.

**Pull request closed without merging**